.gitignore

```diff
@@ -27,3 +29,8 @@ lit.site.cfg
 autoconf/aclocal.m4
 autoconf/autom4te.cache/
 autom4te.cache/
+
+# cmake files
+/CMakeCache.txt
+CMakeFiles
+cmake_install.cmake
```

Comment:
Ouch no. Please disallow in source builds explicitly in the root `CMakeLists.txt` so we can stop people from being silly.